Revise 'Kubernetes Components' page (#40129)
kubernetes/website: content/en/docs/reference/glossary/kube-scheduler.md

tags:
- architecture
---

> Control plane component that watches 
> for newly created {{< glossary_tooltip term_id="pod" text="Pods" >}} 
> that have not yet been assigned to any {{< glossary_tooltip term_id="node" text="node">}}

Reviewer: ```suggestion
A control plane component that watches 
for new, unassigned {{< glossary_tooltip term_id="pod" text="Pods" >}} 
and chooses a {{< glossary_tooltip term_id="node" text="node">}} for them.
```

Author: suggested changes done.

tags:
- architecture
---

> A control plane component that watches 

Reviewer 2: If we are improving the longer versions of the glossary, I think it also makes sense to update the `short_description` in the front matter at the top of the file to use the same wording.

for new unassigned {{< glossary_tooltip term_id="pod" text="Pods" >}} 
and chooses a {{< glossary_tooltip term_id="node" text="node">}} for them.

<!--more-->

Factors taken into account for scheduling decisions include:

- individual and collective resource requirements
- hardware/software/policy constraints
- affinity and anti-affinity specifications
- data locality
- inter-workload interference
- deadlines

> Individual and collective resource requirements: The CPU and memory requirements 

Reviewer 2: @KhushPatibandha I like the bulleted list, but I think the rest might be too much information for the glossary section. I'd suggest we add a link to a page where we talked about these in more detail instead. Maybe this page, https://kubernetes.io/docs/concepts/scheduling-eviction/kube-scheduler/ 

And if we dont have this information written out in the docs yet, and you think it should still be added, please consider creating a new PR adding the content to kube-scheduler page

Reviewer: @KhushPatibandha, I'm sorry I lost the thread on this, but agree with @a-mccarthy's feedback here.
